proposals/nnnn-app-lib-vehicle-data-manager.md

```diff
@@ -0,0 +1,215 @@
+
+/// Requests updated data from the head unit for some vehicle data types. If the data is already subscribed by some observer, the cached data is up to date, so no request will be sent for that type.
+/// @param type The vehicle data types to request.
+/// @param handler The handler to be called when the vehicle data has finished updating.
+- (void)updateVehicleDataTypes:(NSArray<SDLVehicleDataType> *)types completionHandler:(SDLVehicleDataUpdateHandler)handler;
+```
+
+#### Java Suite
+##### Cached Data
+The Java Suite handles cached data differently than the iOS and JavaScript Suite libraries. They provide a single method and rely on the developer to cast the result.
+
+```java
+public Object getCapability(final VehicleDataType type);
+```
+
+##### Retrieving Updated Data
+```java
+public interface OnVehicleDataListener {
+    void onVehicleDataChanged(@NonNull VehicleDataType type, VehicleDataManager manager);
+    void onVehicleDataError(String info);
+}
+
+public void updateCapabilities(final List<VehicleDataType> types, final OnVehicleDataListener listener);
+```
+
+#### JavaScript Suite
+##### Cached Data
+The cached data will be similar to the iOS version of cached data above.
+
+##### Retrieving Updated Data
+```js
+async updateCapability (vehicleDataType)
+```
+
+### Subscribing to Data
+When data is subscribed, the developer will receive updates on a regular basis as the vehicle data value updates. It uses the `SubscribeVehicleData` and `UnsubscribeVehicleData` RPCs under the hood.
+
+#### iOS
+```objc
+/// A handler to be called when vehicle data unsubscribes
+///
+/// @param error An error if one occurred while unsubscribing
+typedef void (^SDLVehicleDataUnsubscribeHandler)(NSError *_nullable error);
+
+/// Subscribes to one or more pieces of vehicle data and calls a handler when any of the types is updated. It will call the handler individually for each update. Vehicle data will be subscribed individually under-the-hood, meaning that the if an error is called, it only relates to that one vehicle data type, but that the callback may be called multiple times with multiple errors.
+///
+/// @param types The vehicle data types to subscribe to for this handler
+/// @param handler The handler to call every time any of the specified vehicle data types update
+/// @return A unique object that can be passed to `unsubscribeFromVehicleData:withObserver:`
+- (id<NSObject>)subscribeToVehicleData:(NSArray<SDLVehicleDataType>)types withUpdateHandler:(SDLVehicleDataUpdateHandler)handler;
+
+/// Subscribes to one or more pieces of vehicle data and calls a selector on an observer when any of these types is updated. It will call the handler individually for each update. Vehicle data will be subscribed individually under-the-hood, meaning that the if an error is called, it only relates to that one vehicle data type, but that the callback may be called multiple times with multiple errors.
```

Comment:
```suggestion
/// Subscribes to one or more pieces of vehicle data and calls a selector on an observer when any of these types is updated. It will call the handler individually for each update. Vehicle data will be subscribed individually under the hood, meaning that if an error is called, it only relates to that one vehicle data type, but that the callback may be called multiple times with multiple errors.
```